### ISSUE TITLE
b-form-group references <label> twice using label-for

### ISSUE THREAD

**Issue by froebera** [bootstrap-vue/bootstrap-vue#2933]
When using `b-form-group` with the `label-for` property, the generated label is also referenced by the group div with the `aria-labelledby` attribute.

This causes ScreenReaders to give out the label multiple times.

Tested with 2.0.0-rc15 & -rc16

https://jsfiddle.net/o9nbhxgr/2/

**Comment by tmorehouse** [comment]
@froebera can you checkout PR #2936 and specifically the preview https://deploy-preview-2936--bootstrap-vue.netlify.com/docs/components/form-group

And see how that works for you?

**Comment by froebera** [comment]
Yes it works for me.

The aria-labelledby is no longer rendered with the `label-for` property and the screenreader output is correct.

But i noticed when onmitting the `label-for` attribute, the fieldset legend is also referenced twice, which also causes the duplicated screenreader output.

**Comment by tmorehouse** [comment]
Yeah, in horizontal mode we add the extra reference due to some browsers not liking the legend not being a direct child of a field set. But maybe browsers (screen readers) are getting better at dealing with non-direct child legends in fieldsets.

**Comment by froebera** [comment]
But isnt the reference at the fieldset level enough even if the screenreader does not support the legend in a sub-child?

JAWS for example does not support nested legends, by its implementation it should give out the legend and the label for each element, which it does not.

But since both the `fieldset` and the content wrapper `div` have the `aria-labelledby` attribute, the legend will be read out twice when a child element recieves focus.

**Comment by tmorehouse** [comment]
Good point...  let me check the code again, to make sure it is only adding the `aria-labelledby` to one element

**Comment by tmorehouse** [comment]
OK, should be fixed now.

https://deploy-preview-2936--bootstrap-vue.netlify.com/docs/components/form-group

`aria-labelledby` is only set if there is no `label-for` provided (meaning a `<fieldset>` with `<legend>`) and the form group is horizontal (meaning at least one of the `label-cols(-{breakpoint})` props is set, and the `<legend>` is no longer a direct child if `<fieldset>` due to the nested `form-row`).

**Comment by froebera** [comment]
Looks good
